packages/bpk-component-calendar/src/date-utils.ts

```diff
@@ -110,7 +115,12 @@ function getCalendarMonthWeeks(date: Date, weekStartsOn: number) {
   for (let i = 0; i < 6; i += 1) {
     const currWeek = [];
     for (let j = 0; j < 7; j += 1) {
-      currWeek.push(dateAtStartOfDay(year, month, day));
+      const currDate = dateAtStartOfDay(year, month, day);
+      currWeek.push({
+        val: currDate,
+        textLabel: formatDate ? formatDate(currDate) : 'loading...',
+        isoLabel: formatIsoDate(currDate),
```

Comment:
Perform all date formatting in a single place, rather than on demand across many files. This decreases the number of date format calls requires and contributes to a 20% reduction in INP.